### DISCUSSION TITLE
Parse mardown from string inside remark plugin

### DISCUSSION THREAD

**Discussion by stevemk14ebr** [remarkjs/remark#725] [Q&A] [answered]
Hello, pretty simple setup i've got. I'm using remark directives as placeholders for some additional markdown that is inserted via a remarkPlugin. This code is pretty simple:

```
function myDerectives() {
    return transform;

    function transform(tree) {
        visit(tree, ['textDirective'], ondirective(tree))
    }

    function ondirective(node) {
        const content = NodeToString(node);
        const tag = node.name;
        if (tag === 'aster') {
            let replacementText = ... do some application specific stuff ...

            // currently using mdast-util-from-markdown
            let renderedChildren = fromMarkdown(rendered);

            // get children of the new root, we don't want a second root that'd be invalid
            node.type = renderedChildren.children[0].type;
            node.children = [...renderedChildren.children[0].children];
        }
    }
}
```

And this works great. HOWEVER, the mdast-util-from-markdown is not going to be configured the with the same plugins as the remark parse that itself invokes my own myDerectives plugin. This becomes a problem if I want my substituted text to also support language extensions like remark-gfm, remark-derictives or anything like that, as i'd have to configure mdast-util-from-markdown to also use those extensions but I can't know which plugins someone uses from within my plugin itself.

So my question. Can I get a reference to the ReMark object itself from a plugin, so that I can parse additional markdown like this, kind of 'recursively' if you will. Ex of what it looks like in my mind:

```
// make our remark object with plugins
const jiraConverter = Remark().use([
        [RemarkGFM, { singleTilde: false, tableCellPadding: false }], RemarkBreaks, RemarkDirective, [myDerective, { schema, renderMap, instanceMap }], RemarkStringify]);

// inside myDerective
let renderedChildren = GetCurrentPipeline().processSync(rendered);
```

The new function would be GetCurrentPipeline() which just gives me the jiraConverter object that itself invoked myDerective. Is this possible?

**Comment by stevemk14ebr** [comment, marked as answer]
Nvm I figured it out. Cool :)

```
function myDerectives() {
    let RemarkParser = this;
    return (...args) => { transform(RemarkParser, ...args) } ;

    function transform(RemrkParser, tree) {
        visit(tree, ['textDirective'], (...args) => { ondirective(RemarkParser, ...args) } )
    }

    function ondirective(RemarkParser, node) {
        const content = NodeToString(node);
        const tag = node.name;
        if (tag === 'aster') {
            let replacementText = ... do some application specific stuff ...

            // currently using mdast-util-from-markdown
            let renderedChildren = RemarkParser.parse(replacementText);

            // get children of the new root, we don't want a second root that'd be invalid
            node.type = renderedChildren.children[0].type;
            node.children = [...renderedChildren.children[0].children];
        }
    }
}
```

**Reply by wooorm** [reply]
This will probably work for your use case.
However, pipelines could be made to start with HTML. Or, the other way around, to *end* with HTML.
So `this` might not completely match with what you expect
